src/indexed_merkle_tree.rs

```diff
@@ -262,16 +256,20 @@ pub fn insert_leaf<F: BigPrimeField, const T: usize, const RATE: usize>(
         is_new_leaf_largest,
     );
 
-    let newlowleaf = IndexedMerkleTreeLeaf {
+    let new_low_leaf = IndexedMerkleTreeLeaf {
         val: low_leaf.val,
-        next_idx: *new_leaf_index,
         next_val: new_leaf.val,
+        next_idx: *new_leaf_index,
     };
 
     let new_low_leaf_hash = hasher.hash_fix_len_array(
         ctx,
         gate,
-        &[newlowleaf.val, newlowleaf.next_val, newlowleaf.next_idx],
+        &[
+            new_low_leaf.val,
+            new_low_leaf.next_val,
+            new_low_leaf.next_idx,
+        ],
     );
 
     let interim_root = compute_merkle_root(
```

Comment:
why have you replaced the `zero_leaf_hash`, `new_leaf_proof` , `new_leaf_proof_helper` with  `new_low_leaf_hash`,`low_leaf_proof`,`low_leaf_proof_helper` ? 
Can you once revisit the logic and let me know what are changes are to be made 
You need to make changes in how the parent hash is calculated using the child nodes and not in how the intermediate roots are calculated